src/java.base/share/classes/java/lang/Module.java

```diff
@@ -256,25 +258,74 @@ public ModuleLayer getLayer() {
     /**
      * Update this module to allow access to restricted methods.
      */
-    Module implAddEnableNativeAccess() {
+    synchronized Module implAddEnableNativeAccess() {
         enableNativeAccess = true;
         return this;
     }
 
     /**
-     * Update all unnamed modules to allow access to restricted methods.
+     * Returns {@code true} if this module can access
+     * <a href="foreign/package-summary.html#restricted"><em>restricted</em></a> methods.
+     *
+     * @return {@code true} if this module can access <em>restricted</em> methods.
      */
-    static void implAddEnableNativeAccessAllUnnamed() {
-        ALL_UNNAMED_MODULE.enableNativeAccess = true;
+    @PreviewFeature(feature=PreviewFeature.Feature.FOREIGN)
+    public boolean isNativeAccessEnabled() {
+        Module target = enableNativeAccessHolder();
+        synchronized(target) {
+            return target.enableNativeAccess;
+        }
+    }
+
+    // Returns the Module object holds the enableNativeAccess
```

Comment:
There's a typo, should be "that holds".  Also the method name is a bit misleading as it doesn't enable anything.  The isNativeAccessEnabled and enableNativeAccess are deeply coupled so the method isn't strictly needed. If you keep it then something like moduleForNativeAccess() in the usage by both methods might be clearer.